File: 253-meeting-rooms-ii/meeting-rooms-ii.cpp
class Solution {
public:
    int minMeetingRooms(vector<vector<int>>& intervals) {
        map<int, int> line;
        
        for (auto interval : intervals) {
            line[interval[0]]++;
            line[interval[1]]--;
        }
        
        int res = 0;
        int count = 0;
        
        for (auto l : line) {
            count += l.second;
            res = max(res, count);
        }
        
        return res;
        
    }
};

File: 253-meeting-rooms-ii/meeting-rooms-ii.java
class Solution {
    public int minMeetingRooms(int[][] intervals) {
        Map<Integer, Integer> line = new TreeMap<>();
        for (int[] interval : intervals) {
            int curr = line.getOrDefault(interval[0], 0);
            line.put(interval[0], curr + 1);

            curr = line.getOrDefault(interval[1], 0);
            line.put(interval[1], curr - 1);
        }

        int count = 0;
        int res = 0;

        for (Integer l : line.keySet()) {
            count += line.get(l);
            res = Math.max(count, res);
        }
        return res;
    }
}